diesel/Cargo.toml

```diff
@@ -55,12 +55,14 @@ huge-tables = ["64-column-tables"]
 postgres = ["pq-sys", "postgres_backend"]
 sqlite = ["libsqlite3-sys", "diesel_derives/sqlite"]
 mysql = ["mysqlclient-sys", "url", "percent-encoding", "bitflags", "mysql_backend"]
-without-deprecated = []
-with-deprecated = []
+without-deprecated = ["diesel_derives/without-deprecated"]
+with-deprecated = ["diesel_derives/with-deprecated"]
 network-address = ["ipnetwork", "libc"]
 numeric = ["num-bigint", "bigdecimal", "num-traits", "num-integer"]
 postgres_backend = ["diesel_derives/postgres", "bitflags", "byteorder"]
 mysql_backend = ["diesel_derives/mysql", "byteorder"]
+i-implement-a-third-party-backend-and-opt-into-breaking-changes = []
```

Comment:
This seems somewhat dangerous. If I depend on such a backend, then I can also access those unstable features, and might use one without realizing.

Reply:
The probably best other solution here is to use a raw cfg flag, right?
The other option that I see is to split up diesel itself into multiple crates and reexport only the stable api via a diesel crate. Everything else is then only exposed via a 0.x internal crate. I assume that this solution will cause problems as well...

Reply:
Well, a raw cfg flag is how Tokio handles unstable features, but it does have the disadvantage that a crate cannot specify it, and that it must be specified by the person running `cargo build`. (But that's also its advantage and why Tokio uses it over a feature.)